fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java

```diff
@@ -154,6 +149,23 @@ public static ExecPlan plan(StatementBase stmt, ConnectContext session,
         return null;
     }
 
+    public static ExecPlan planInsertStmt(PlannerMetaLocker plannerMetaLocker,
+                                          InsertStmt insertStmt,
+                                          ConnectContext connectContext) {
+        // if use optimistic lock, we will unlock it in InsertPlanner#buildExecPlanWithRetrye
+        boolean useOptimisticLock = isLockFreeInsertStmt(insertStmt, connectContext);
+        return new InsertPlanner(plannerMetaLocker, useOptimisticLock).plan(insertStmt, connectContext);
+    }
+
+    private static boolean isLockFreeInsertStmt(InsertStmt insertStmt,
+                                                ConnectContext connectContext) {
+        boolean isSelect = !(insertStmt.getQueryStatement().getQueryRelation() instanceof ValuesRelation);
```

Comment:
why `VALUES` cannot be considered as lockfree ?

Reply:
You can check in this: https://github.com/StarRocks/starrocks/pull/39141

This PR only add a function instead of changing anything.